azurerm/internal/services/sentinel/sentinel_alert_rule_ms_security_incident_resource.go

```diff
@@ -196,28 +196,35 @@ func resourceSentinelAlertRuleMsSecurityIncidentCreateUpdate(d *schema.ResourceD
 	if !d.IsNewResource() {
 		resp, err := client.Get(ctx, workspaceID.ResourceGroup, "Microsoft.OperationalInsights", workspaceID.WorkspaceName, name)
 		if err != nil {
-			return fmt.Errorf("retrieving Sentinel Alert Rule Ms Security Incident %q (Resource Group %q / Workspace: %q): %+v", name, workspaceID.ResourceGroup, workspaceID.WorkspaceName, err)
+			return fmt.Errorf("retrieving Sentinel Alert Rule Ms Security Incident %q (Workspace: %q): %+v", name, workspaceID, err)
 		}
 
 		if err := assertAlertRuleKind(resp.Value, securityinsight.MicrosoftSecurityIncidentCreation); err != nil {
-			return fmt.Errorf("asserting alert rule of %q (Resource Group %q / Workspace: %q): %+v", name, workspaceID.ResourceGroup, workspaceID.WorkspaceName, err)
+			return fmt.Errorf("asserting alert rule of %q (Workspace: %q): %+v", name, workspaceID, err)
 		}
 		param.Etag = resp.Value.(securityinsight.MicrosoftSecurityIncidentCreationAlertRule).Etag
 	}
 
 	if _, err := client.CreateOrUpdate(ctx, workspaceID.ResourceGroup, "Microsoft.OperationalInsights", workspaceID.WorkspaceName, name, param); err != nil {
-		return fmt.Errorf("creating Sentinel Alert Rule Ms Security Incident %q (Resource Group %q / Workspace: %q): %+v", name, workspaceID.ResourceGroup, workspaceID.WorkspaceName, err)
+		return fmt.Errorf("creating Sentinel Alert Rule Ms Security Incident %q (Workspace: %q): %+v", name, workspaceID, err)
 	}
 
 	resp, err := client.Get(ctx, workspaceID.ResourceGroup, "Microsoft.OperationalInsights", workspaceID.WorkspaceName, name)
 	if err != nil {
-		return fmt.Errorf("retrieving Sentinel Alert Rule Ms Security Incident %q (Resource Group %q / Workspace: %q): %+v", name, workspaceID.ResourceGroup, workspaceID.WorkspaceName, err)
+		return fmt.Errorf("retrieving Sentinel Alert Rule Ms Security Incident %q (Workspace: %q): %+v", name, workspaceID, err)
 	}
+
 	id := alertRuleID(resp.Value)
 	if id == nil || *id == "" {
-		return fmt.Errorf("empty or nil ID returned for Sentinel Alert Rule Ms Security Incident %q (Resource Group %q / Workspace: %q) ID", name, workspaceID.ResourceGroup, workspaceID.WorkspaceName)
+		return fmt.Errorf("empty or nil ID returned for Sentinel Alert Rule Ms Security Incident %q (Workspace: %q) ID", name, workspaceID)
+	}
+
+	resId, err := parse.AlertRuleID(*id)
```

Comment:
(as above) we should be able to compose this information based off user input, rather than making this additional Get request and parsing this from the API